graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -213,28 +190,47 @@ class VertexType(GraphElement):
     def __init__(self, class_name, abstract, properties, class_fields):
         super(VertexType, self).__init__(class_name, abstract, properties, class_fields)
 
-        # Non-edges must not have properties like "in" or "out" defined, and
-        # must not have properties of type "Link".
-        _validate_non_edges_do_not_have_edge_like_properties(class_name, properties)
-
 
 class EdgeType(GraphElement):
-    def __init__(self, class_name, abstract, properties, class_fields):
-        super(EdgeType, self).__init__(class_name, abstract, properties, class_fields)
+    EDGE_SOURCE_PROPERTY_NAME = 'out'
+    EDGE_DESTINATION_PROPERTY_NAME = 'in'
+    EDGE_END_NAMES = {EDGE_SOURCE_PROPERTY_NAME, EDGE_DESTINATION_PROPERTY_NAME}
+
```

Comment:
Moved EDGE_SOURCE_PROPERTY_NAME and EDGE_DESTINATION_PROPERTY_NAME here because these fields are idiosyncratic requirements that we have on link names and I think I might rename the schema_properties file to orientdb_schema_properties at some point in the future.